src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -585,25 +601,25 @@ private void processLogsAndUpload() throws InterruptedException {
                             }
                         }
                     }
+                    // If there are no log files for the component, then return.
+                    if (allFiles.isEmpty()) {
+                        continue;
+                    }
                     // Sort the files by the last modified time. Then try to proceed oldest file first to avoid data
                     // loss caused by auto deletion
                     allFiles.sort(Comparator.comparingLong(LogFile::lastModified));
```

Comment:
why do we need any of this logic anymore? We want to treat all files exactly the same now, don't we?

Reply:
If this is only used for deletion, then it should be filtered in the deletion logic. This logic doesn't sound like it is required right here.